ui/main/src/app/store/effects/card-operation.effects.ts

```diff
@@ -49,26 +51,47 @@ export class CardOperationEffects {
             map((loadedCardAction: LoadLightCard) => loadedCardAction.payload.lightCard),
             withLatestFrom(this.store.select(selectLightCardsState)),
             map(([lightCard, state]) => {
-                    if (!!lightCard.parentCardId) {
-                        return new LoadLightChildCard({lightCard: lightCard,isFromCurrentUserEntity : this.doesChildCardIsFromCurrentUserEntity(lightCard)});
-                    }
-                    else {
-                        let card: LightCard  = {...lightCard} ;
-                        card.hasChildCardFromCurrentUserEntity = this.doesNewCardVersionHasChildFromCurrentUserEntity(state.entities[card.id],card);
-                        return new LoadLightParentCard({lightCard: card});
+                if (!!lightCard.parentCardId) {
+                    const isFromCurrentUser = this.doesChildCardIsFromCurrentUserEntity(lightCard);
+                    if (isFromCurrentUser) {
+                        if (!state.entities[lightCard.parentCardId]) {
+                             // if parent does not exist yet keep in memory the information that the card with id=parentCardId has a child
+                             // and that this child if from current user entity
+                             // this situation can arise as we do not have the card ordered when received from
+                             // the back via subscription (i.e a child can be received  before his parent)
+                            this.orphanedChildCardsFromCurrentEntity.add(lightCard.parentCardId);
+                            // do not need to load this child card
+                            return new NoopAction();
+                        }
+    
                     }
+                    return new LoadLightChildCard({lightCard: lightCard, isFromCurrentUserEntity: isFromCurrentUser});
```

Comment:
I didn't manage to understand why the orphaned child card situation should only be handled in the case it is from the current user?

Reply:
The only use of child card in the state is when you have a detail card open and a child for this card arrives (the detail card get the child form the state via the lastCardLoaded value) . 

In the previous code , the parent card is not loaded so it is not open in the UI , so we do not need to had the child in the store